encoder/vaapiencoder_hevc.cpp

```diff
@@ -1290,7 +1290,7 @@ void VaapiEncoderHEVC::shortRfsUpdate(const PicturePtr& picture)
     m_shortRFS.inter_ref_pic_set_prediction_flag = 0;
 
     if (intraPeriod() > 1 && m_refList0.size()) {
-        m_shortRFS.num_negative_pics         = 1;
+        m_shortRFS.num_negative_pics         = m_refList0.size();
```

Comment:
yes, it is my fault, it should be only set for negative pics. You can discard my patch and fix it with another patch